src/frontend/src/utils/utils.ts

```diff
@@ -1,3 +1,20 @@
+import { Ref } from "lit-html/directives/ref.js";
```

Comment:
not sure about this import, you might have issue someday if you switch to a module based bundler - just a feeling though